Jenkinsfile

```diff
@@ -1,25 +1,35 @@
+// Overall pipeline looks like the following
+//
+// |--Pre-build jobs--|----BUILD----|--Pre-Coverage--|------Test------|-Post Coverage-|--Build rest--|
+//                      |             |                |                |                |
+//                      |-> Linux     |-> lcov         |-> Linux        |-> lcov         |-> gen docs
+//                      |-> ARMv7                      |-> ARMv7        |-> SonarQube    |-> bindings
+//                      |-> ARMv8                      |-> ARMv8
+//                      |-> MacOS                      |-> MacOS
+//
+// NOTE: In build stage we differentiate only platforms in pipeline scheme. Build/Release is filtered inside the platform
+
 properties([parameters([
-  booleanParam(defaultValue: true, description: 'Build `iroha`', name: 'iroha'),
-  booleanParam(defaultValue: false, description: 'Build `bindings`', name: 'bindings'),
-  booleanParam(defaultValue: true, description: '', name: 'x86_64_linux'),
-  booleanParam(defaultValue: false, description: '', name: 'armv7_linux'),
-  booleanParam(defaultValue: false, description: '', name: 'armv8_linux'),
-  booleanParam(defaultValue: true, description: '', name: 'x86_64_macos'),
-  booleanParam(defaultValue: false, description: '', name: 'x86_64_win'),
-  choice(choices: 'Debug\nRelease', description: 'Iroha build type', name: 'build_type'),
-  booleanParam(defaultValue: false, description: 'Build Java bindings', name: 'JavaBindings'),
-  choice(choices: 'Release\nDebug', description: 'Java bindings build type', name: 'JBBuildType'),
-  booleanParam(defaultValue: false, description: 'Build Python bindings', name: 'PythonBindings'),
-  choice(choices: 'Release\nDebug', description: 'Python bindings build type', name: 'PBBuildType'),
-  choice(choices: 'python3\npython2', description: 'Python bindings version', name: 'PBVersion'),
-  booleanParam(defaultValue: false, description: 'Build Android bindings', name: 'AndroidBindings'),
+  choice(choices: 'Debug\nRelease', description: '', name: 'BUILD_TYPE'),
+  booleanParam(defaultValue: true, description: '', name: 'Linux'),
+  booleanParam(defaultValue: false, description: '', name: 'ARMv7'),
+  booleanParam(defaultValue: false, description: '', name: 'ARMv8'),
+  booleanParam(defaultValue: false, description: '', name: 'MacOS'),
+  booleanParam(defaultValue: false, description: 'Whether it is a triggered build', name: 'Nightly'),
```

Comment:
I would remove "whether" from titles, because the switches themselves imply this word, so it does not add any information.

Reply:
Removed